data/blog/from-native-to-js.md

```diff
@@ -0,0 +1,103 @@
+---
+title: From Native to JS in Electron
+author: codebytere
+date: '2018-03-19'
+---
+
+How do Electron's features written in C++ or Objective-C get to JavaScript so they're available to an end-user?
+
+---
+
+## Background
+
+[Electron](https://electronjs.org) is a JavaScript platform whose primary purpose is to lower the barrier to entry for developers to build robust desktop apps. However, at its core, Electron doesn't remove the need for platform-specific functionality to be written in a given system language.
+
+In reality, Electron handles the native code for you so that you can focus on a single JavaScript API.
+
+How does that work, though? How do Electron's features written in C++ or Objective-C get to JavaScript so they're available to an end-user?
+
+To trace this pathway, start with the [`app` module](https://electronjs.org/docs/api/app).
+
+By opening the [`app.ts`](https://github.com/electron/electron/tree/master/lib/browser/api/app.ts) file inside our `lib/` directory, you'll find the following line of code towards the top:
```

Comment:
Maybe link to an exact commit hash here so that the link makes sense for future visitors

Reply:
Also probably worth noting somewhere in this post that this is The Way Things Works as of now and could change in the future if you're reading this blog post a year on 👍